### PR TITLE
tests, executor, br: fix time_zone related bugs in UT

### LINKED ISSUE #49493: Some tests can't pass when time_zone not equal to `Asia/Shanghai`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Some tests and feature can't pass when `time_zone != Asia/Shanghait`, we should fix it.

You could get list of test cases below

- [ ] TestStmtFile 
- [x] TestFlashbackCloseAndResetPDSchedule 
- [x] TestTiDBTrx 
- [ ] TestExactStalenessTransaction 
- [x] TestSetTransactionReadOnlyAsOf 
- [x] TestCancelFlashbackCluster 
- [ ] TestSetTiDBCloudStorageURI 
- [ ] TestIssue20236 
- [ ] TestClusterTableSlowQuery 
- [ ] TestGetLock 
- [ ] TestSelectClusterTable 
- [x] TestTiFlashFlashbackCluster 
- [x] TestGlobalVariablesOnFlashback 
- [ ] TestStmtFiles 
- [ ] TestChangingColOriginDefaultValueAfterAddColAndCastSucc 
- [x] TestAddDDLDuringFlashback 
- [ ] TestStmtFileInvalidLine 
- [x] [TestGetTimeZone](https://github.com/pingcap/tidb/issues/42755) 
- [x] TestRecoverClusterMeetError 

[log.txt](https://github.com/pingcap/tidb/files/13681847/log.txt)

### 2. What did you expect to see? (Required)

The test shouldn't related with time_zone.

### 3. What did you see instead (Required)

The test failed.

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #49493

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/integration_test/integration_test.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func TestGetLock(t *testing.T) {
 	// Increase pessimistic txn max retry count to make test more stable.
 	originCfg := config.GetGlobalConfig()
 	newCfg := *originCfg
-	newCfg.PessimisticTxn.MaxRetryCount = 2048
+	newCfg.PessimisticTxn.MaxRetryCount = 10240
```
Comment (zimulala) [inline pkg/expression/integration_test/integration_test.go:66]: Why change this value?

Comment (Defined2014) [reply pkg/expression/integration_test/integration_test.go:66]: As `Increase pessimistic txn max retry count to make test more stable.`, but sometimes `2048` is not big enough.

**Review by zimulala — APPROVED** [review]

File: br/pkg/task/show/cmd_test.go [hunk 2]
```diff
@@ -178,6 +178,7 @@ func TestShowViaSQL(t *testing.T) {
 	err := os.WriteFile(metaPath, FullMeta, 0o444)
 	req.NoError(err)
 
+	tk.MustExec("set @@time_zone='+08:00'")
```
Comment (BornChanger) [inline br/pkg/task/show/cmd_test.go:181]: please also add other time_zone to verify the output will change.

Comment (Defined2014) [reply br/pkg/task/show/cmd_test.go:181]: Done

**Review by XuHuaiyu — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**